bin/host/src/fetcher/mod.rs

```diff
@@ -182,11 +185,132 @@ where
+                        .request::<&[B256; 1], Bytes>("debug_dbGet", &[hash])
+                        .await
+                        .map_err(|e| anyhow!("Error fetching code hash preimage: {e}"))?,
+                };
+
+                let mut kv_write_lock = self.kv_store.write().await;
+                kv_write_lock.set(hash, code.into());
+            }
+            HintType::StartingL2Output => {
+                const OUTPUT_ROOT_VERSION: u8 = 0;
+                const L2_TO_L1_MESSAGE_PASSER_ADDRESS: Address =
+                    address!("4200000000000000000000000000000000000016");
+
+                if !hint_data.is_empty() {
+                    anyhow::bail!("Invalid hint data length: {}", hint_data.len());
+                }
+
+                // Fetch the header for the L2 head block.
+                let raw_header: Bytes = self
+                    .l2_provider
+                    .client()
+                    .request("debug_getRawHeader", &[self.l2_head])
+                    .await
+                    .map_err(|e| anyhow!("Failed to fetch header RLP: {e}"))?;
+                let header = Header::decode(&mut raw_header.as_ref())
+                    .map_err(|e| anyhow!("Failed to decode header: {e}"))?;
+
+                // Fetch the storage root for the L2 head block.
+                let l2_to_l1_message_passer = self
+                    .l2_provider
+                    .get_proof(
+                        L2_TO_L1_MESSAGE_PASSER_ADDRESS,
+                        Default::default(),
+                        self.l2_head.into(),
+                    )
+                    .await
+                    .map_err(|e| anyhow!("Failed to fetch account proof: {e}"))?;
+
+                let mut raw_output = [0u8; 97];
+                raw_output[0] = OUTPUT_ROOT_VERSION;
+                raw_output[1..33].copy_from_slice(header.state_root.as_ref());
+                raw_output[33..65].copy_from_slice(l2_to_l1_message_passer.storage_hash.as_ref());
+                raw_output[65..97].copy_from_slice(self.l2_head.as_ref());
+                let output_root = keccak256(raw_output);
+
+                let mut kv_write_lock = self.kv_store.write().await;
+                kv_write_lock.set(output_root, raw_output.into());
+            }
+            HintType::L2StateNode => {
+                if hint_data.len() != 32 {
+                    anyhow::bail!("Invalid hint data length: {}", hint_data.len());
+                }
+
+                let hash: B256 = hint_data
+                    .as_ref()
+                    .try_into()
+                    .map_err(|e| anyhow!("Failed to convert bytes to B256: {e}"))?;
+
+                // Fetch the preimage from the L2 chain provider.
+                // TODO(clabby): Unfortunately, we rely on hash scheme geth's `debug_dbGet` to
+                // retrieve the preimage. This is not ideal and should be replaced
+                // with a more robust solution in the future.
```

Comment:
Can we make this and the `L2Code` hint type's `debug_dbGet` use above into tickets to track better?

Strong preference for issues over inline TODO comments for visibility.